applications/RANSApplication/custom_strategies/algebraic_flux_corrected_steady_scalar_scheme.h

```diff
@@ -0,0 +1,582 @@
+            << std::scientific << mRelaxationFactor << "\n";
+
+        mpDofUpdater = Kratos::make_unique<DofUpdaterType>(mRelaxationFactor);
+    }
+
+    ~AlgebraicFluxCorrectedSteadyScalarScheme() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        BaseType::Initialize(rModelPart);
+
+        if (mrPeriodicIdVar != Variable<int>::StaticObject()) {
+            block_for_each(rModelPart.Conditions(), [&](const ModelPart::ConditionType& rCondition) {
+                if (rCondition.Is(PERIODIC)) {
+                    // this only supports 2 noded periodic conditions
+                    KRATOS_ERROR_IF(rCondition.GetGeometry().PointsNumber() != 2)
+                        << this->Info() << " only supports two noded periodic conditions. Found "
+                        << rCondition.Info() << " with "
+                        << rCondition.GetGeometry().PointsNumber() << " nodes.\n";
+
+                    const auto& r_node_0 = rCondition.GetGeometry()[0];
+                    const std::size_t r_node_0_pair_id =
+                        r_node_0.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    const auto& r_node_1 = rCondition.GetGeometry()[1];
+                    const std::size_t r_node_1_pair_id =
+                        r_node_1.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    KRATOS_ERROR_IF(r_node_0_pair_id != r_node_1.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_0_pair_id
+                        << " != " << r_node_1.Id() << " ].\n";
+
+                    KRATOS_ERROR_IF(r_node_1_pair_id != r_node_0.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_1_pair_id
+                        << " != " << r_node_0.Id() << " ].\n";
+                }
+            });
+        }
+
+        // Allocate auxiliary memory.
+        const auto num_threads = OpenMPUtils::GetNumThreads();
+        mAntiDiffusiveFlux.resize(num_threads);
+        mAntiDiffusiveFluxCoefficients.resize(num_threads);
+        mValues.resize(num_threads);
+        mAuxMatrix.resize(num_threads);
+
+        KRATOS_CATCH("");
+    }
+
+    void InitializeNonLinIteration(
+        ModelPart& rModelPart,
+        TSystemMatrixType& A,
+        TSystemVectorType& Dx,
+        TSystemVectorType& b) override
+    {
+        KRATOS_TRY
+
+        auto& r_nodes = rModelPart.Nodes();
+
+        block_for_each(r_nodes, [&](ModelPart::NodeType& rNode) {
+            rNode.SetValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX, 0.0);
+            rNode.SetValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX, 0.0);
+            rNode.SetValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT, 0.0);
+            rNode.SetValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT, 0.0);
```

Comment:
```suggestion
            rNode.GetValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX) = 0.0;
            rNode.GetValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX) = 0.0;
            rNode.GetValue(AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT) = 0.0;
            rNode.GetValue(AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT) = 0.0;
```
And mover the `SetValue` to the `Initialize` to make it threadsafe.

Reply:
done :)